fml/shared_thread_merger_impl.cc

```diff
@@ -0,0 +1,110 @@
+// Copyright 2013 The Flutter Authors. All rights reserved.
+// Use of this source code is governed by a BSD-style license that can be
+// found in the LICENSE file.
+
+#define FML_USED_ON_EMBEDDER
+
+#include "flutter/fml/shared_thread_merger_impl.h"
+
+#include <set>
+#include "flutter/fml/message_loop_impl.h"
+
+namespace fml {
+
+std::mutex SharedThreadMergerImpl::creation_mutex_;
+
+// Guarded by creation_mutex_
+std::map<ThreadMergerKey, fml::RefPtr<SharedThreadMergerImpl>>
+    SharedThreadMergerImpl::shared_merger_instances_;
```

Comment:
https://google.github.io/styleguide/cppguide.html#Static_and_Global_Variables

Objects with nontrivial destructors aren't allowed for static variables.  Maybe you can push this concept one higher to where we have a shared object?  I'd don't have a good suggestion off the top of my head, I'd have to dig into it.

You might have to make a shared object for this information that is shared when an engine gets spawned from another one.

Reply:
Maybe I can use the raw pointer? `std::map<ThreadMergerKey, SharedThreadMergerImpl*> shared_merger_instances_;` and calling `new` and  `delete` manually? @gaaclarke @iskakaushik @gooooloo

Reply:
Raw pointers aren't technically allowed either: https://google.github.io/styleguide/cppguide.html#Ownership_and_Smart_Pointers

(I made an exception below because you aren't actually calling into the pointers and just using them for identification purposes)

Reply:
So how can I deal with it? let me think more ways about it

Reply:
The information could live in the `Shell` since that is shared between engines and knows about the rasterizer.

Reply:
@gaaclarke The shell is called `new` by `CreateWithSnapshot()` in shell.cc. Maybe I can make the set as a static member of `Shell` class and destroy the item in set when `~Shell()` destructor is called.

Reply:
Done. I put the information in Shell as a member called `parent_raster_thread_merger_`, and the child engine will create a  new `RasterThreaddMerger` instance as caller and shares parent merger object's `shared_thread_merger` member.